web/src/components/tasks/cigarette-license/CigaretteLicenseReview.tsx

```diff
@@ -0,0 +1,411 @@
+import { useFormContextFieldHelpers } from "@/lib/data-hooks/useFormContextFieldHelpers";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { isTradeNameLegalStructureApplicable } from "@/lib/domain-logic/isTradeNameLegalStructureApplicable";
+import { SubmissionError } from "@businessnjgovnavigator/shared/cigaretteLicense";
+import { useRouter } from "next/compat/router";
+import { ReactElement, useContext } from "react";
+
+interface Props {
+  setStepIndex: (step: number) => void;
+  setSubmissionError: (error: SubmissionError) => void;
+  CMS_ONLY_show_error?: boolean;
+}
+
+export const CigaretteLicenseReview = (props: Props): ReactElement => {
+  const { Config } = useConfig();
+  const {
+    state: cigaretteLicenseData,
+    saveCigaretteLicenseData,
+    setCigaretteLicenseData,
+  } = useContext(CigaretteLicenseContext);
+  const { state: profileDataState } = useContext(ProfileDataContext);
+  const { state: addressState } = useContext(AddressContext);
+  const { userData, business } = useUserData();
+  const profileData = profileDataState.profileData;
+  const formationAddressData = addressState.formationAddressData;
+  const router = useRouter();
+
+  const { setIsValid: setIsValidBusinessName } = useFormContextFieldHelpers(
+    "businessName",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidResponsibleOwnerName } = useFormContextFieldHelpers(
+    "responsibleOwnerName",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidTradeName } = useFormContextFieldHelpers(
+    "tradeName",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidTaxId } = useFormContextFieldHelpers(
+    "taxId",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidAddressLine1 } = useFormContextFieldHelpers(
+    "addressLine1",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidAddressCity } = useFormContextFieldHelpers(
+    "addressCity",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidAddressState } = useFormContextFieldHelpers(
+    "addressState",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidAddressZipCode } = useFormContextFieldHelpers(
+    "addressZipCode",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidMailingAddressLine1 } = useFormContextFieldHelpers(
+    "mailingAddressLine1",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidMailingAddressCity } = useFormContextFieldHelpers(
+    "mailingAddressCity",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidMailingAddressState } = useFormContextFieldHelpers(
+    "mailingAddressState",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidMailingAddressZipCode } = useFormContextFieldHelpers(
+    "mailingAddressZipCode",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidSalesInfoStartDate } = useFormContextFieldHelpers(
```

Comment:
For reviewers: Depending on whether or not Step 3 has been merged (and this subsequently rebased), you may not be able to complete the form through the application UI. You can temporarily comment out these lines for Sales Info Start Date and Sales Info Supplier in order to test the submission functionality in the meantime.